src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -296,35 +315,47 @@ public static string GetMsbuildDirectory(string version, IConsole console)
             }
             else
             {
-                var versionString = version.Contains('.') ?
-                    version :
-                    version + ".0";
-                Version ver;
-                if (!Version.TryParse(versionString, out ver))
-                {
-                    var message = string.Format(
-                        CultureInfo.CurrentCulture,
-                        LocalizedResourceManager.GetString(
-                            nameof(NuGetResources.Error_InvalidMsbuildVersion)),
-                        version);
+                // append ".0" if the userVersion is a number
+                string userVersionString = userVersion;
+                int unused;
 
-                    throw new CommandLineException(message);
+                if (int.TryParse(userVersion, out unused))
+                {
+                    userVersionString = userVersion + ".0";
                 }
 
+                Version ver;
+                bool hasNumericVersion = Version.TryParse(userVersionString, out ver);
+
                 var selectedToolset = installedToolsets.FirstOrDefault(
-                    toolset =>
+                toolset =>
+                {
+                    // first match by string comparison
+                    if (string.Equals(userVersionString, toolset.ToolsVersion, StringComparison.OrdinalIgnoreCase))
                     {
-                        var toolsVersion = new Version(toolset.ToolsVersion);
+                        return true;
+                    }
+
+                    if (hasNumericVersion)
+                    {
+                        // then match by Major & Minor version numbers.
+                        var toolsVersion = SafeParseVersion(toolset.ToolsVersion);
+
                         return (toolsVersion.Major == ver.Major &&
```

Comment:
You're right. Fixed so major & minor numbers match is used only when both the userVersion and toolset.ToolsVersion are valid Versions.